include/deal.II/arborx/access_traits.h

```diff
@@ -0,0 +1,262 @@
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_arborx_access_traits_h
+#define dealii_arborx_access_traits_h
+
+#include <deal.II/base/config.h>
+
+#ifdef DEAL_II_WITH_ARBORX
+#  include <deal.II/base/bounding_box.h>
+
+#  include <ArborX.hpp>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ *  Structure that encapsulates a vector of dealii::Points used as queries.
+ */
+struct PointIntersect
+{
+  /**
+   * Constructor.
+   */
+  template <int dim, typename Number>
+  PointIntersect(std::vector<dealii::Point<dim, Number>> const &dim_points)
+  {
+    static_assert(dim != 1, "dim equal to one is not supported.");
+
+    unsigned int const size = dim_points.size();
+    points.reserve(size);
+    for (unsigned int i = 0; i < size; ++i)
+      {
+        points.emplace_back(static_cast<float>(dim_points[i][0]),
+                            static_cast<float>(dim_points[i][1]),
+                            dim == 2 ? 0. :
+                                       static_cast<float>(dim_points[i][2]));
+      }
+  }
+
+  /**
+   * Number of Point stored in the structure.
+   */
+  std::size_t
+  size() const
+  {
+    return points.size();
+  }
+
+  std::vector<dealii::Point<3, float>> points;
+};
+
+
+
+/**
+ * Structure that encapsulates a vector of dealii::BoundingBox used as
+ * queries.
+ */
+struct BoundingBoxIntersect
+{
+  /**
+   * Constructor.
+   */
+  template <int dim, typename Number>
+  BoundingBoxIntersect(std::vector<dealii::BoundingBox<dim, Number>> const &bb)
+  {
+    unsigned int const size = bb.size();
+    bounding_boxes.reserve(size);
+    dealii::Point<3, float> min_corner_arborx(0., 0., 0.);
+    dealii::Point<3, float> max_corner_arborx(0., 0., 0.);
+    for (unsigned int i = 0; i < size; ++i)
+      {
+        auto boundary_points                  = bb[i].get_boundary_points();
+        dealii::Point<dim, Number> min_corner = boundary_points.first;
+        dealii::Point<dim, Number> max_corner = boundary_points.second;
+        for (int d = 0; d < dim; ++d)
+          {
+            min_corner_arborx[d] = static_cast<float>(min_corner[d]);
+            max_corner_arborx[d] = static_cast<float>(max_corner[d]);
+          }
+        bounding_boxes.emplace_back(
+          std::make_pair(min_corner_arborx, max_corner_arborx));
+      }
+  }
+
+  /**
+   * Number of Point stored in the structure.
+   */
+  std::size_t
+  size() const
+  {
+    return bounding_boxes.size();
+  }
+
+  std::vector<dealii::BoundingBox<3, float>> bounding_boxes;
+};
+
+DEAL_II_NAMESPACE_CLOSE
+
+namespace ArborX
```

Comment:
can you add a single line comment that then shows up in our doxygen documentation, explaining what this namespace is there for?